### ISSUE TITLE
Consider moving bitmask definitions into desiutil

### ISSUE THREAD

**Issue by weaverba137** [desihub/desiutil#113]
Currently, desiutil contains the *code* to manipulate DESI bitmasks, but it does not contain the bitmask *data*.  In fact, the bitmask data seems to be scattered around several different packages, and this does not even consider the imaging survey bitmasks.

In SDSS, all bitmask definitions are contained in one place, in the idlutils product.  For DESI, we don't necessarily need to put bitmask definitions into one *file*, but it would be nice to have them all in one *package*.

**Comment by tskisner** [comment]
I agree that bit definitions for a given "thing" should exist in only one package.  However, one thing to consider is that it might make sense to have different "things" in different packages (close to the code which "sets" those bits).  

For example, the target bitmasks are defined in one package (desitarget), and people familiar with targeting can tweak those without touching other packages.  For bit definitions related to the instrument (e.g. fiberstatus), perhaps those should go with the other focalplane / hardware tools in desimodel?

More examples:  if there were bit definitions associated with the quality of pipeline processing, it might make sense to put those in desispec.  If there were bit definitions associated with the quality of the redshift fitting, then those might go in redrock.

In all cases though the bit definitions should not be **duplicated** anywhere.

If we start putting some of these bit definitions in desiutil rather than "with the code" that primarily sets those bits, then I guess it would make sense to move *everything* to desiutil.  There are pros and cons to both approaches, but I definitely agree that a given bit should only be defined in one place.

**Comment by weaverba137** [comment]
Thanx @tskisner , I agree with all of that, but there's another case to consider: downstream analysis.  Having bitmasks in several packages creates dependencies on all of those packages for anyone who needs bitmasks in analysis code.  Maybe we're OK with that and say "you've got to use the entire DESI software stack for analysis".  But that should be discussed.

**Comment by weaverba137** [comment]
No action right now, but once DECaLS DR9 is done, add the final imaging bitmasks to desiutil.  On the longer-term, if, *e.g.* desispec bitmasks are ever finalized, those could be moved into desiutil.  There is some debate as to whether desitarget bitmasks will *ever* be finalized.

**Comment by weaverba137** [comment]
@sbailey, I recommend closing this issue in favor of simply documenting where all of the mask bit definitions are, *i.e.* desihub/desidatamodel#118.

**Comment by sbailey** [comment]
Agreed.  We'll keep the bit definitions where they are currently, and focus on documenting them in the data model.
